### PR TITLE
Fix tooltip for ReferenceExpr in lambda analyzer warning and quick fix

### PR DESCRIPTION
**Pull request by DedSec256**
https://youtrack.jetbrains.com/issue/RIDER-51461

### CODE REVIEW

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Highlightings/FSharpErrorUtil.fs [hunk 1]
```diff
@@ -96,9 +96,12 @@ let getReferenceExprName (expr: IFSharpExpression) =
     | :? IReferenceExpr as refExpr -> refExpr.ShortName
     | _ -> SharedImplUtil.MISSING_DECLARATION_NAME
 
+let qualifiedByReference (expr: IReferenceExpr) = not expr.IsQualified || expr.Qualifier :? IReferenceExpr
+
 let getLambdaCanBeReplacedWarningText (replaceCandidate: IFSharpExpression) =
     match replaceCandidate with
-    | :? IReferenceExpr as x -> sprintf "Lambda can be replaced with '%s'" x.QualifiedName
+    | :? IReferenceExpr as x when qualifiedByReference x ->
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Highlightings/FSharpErrorUtil.fs:103]: Doesn't work for simple cases:
```fsharp
"".Foo.Bar
```

File: ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/ReferenceExpr.cs [hunk 2]
```diff
@@ -82,6 +82,7 @@ public FSharpSymbolReference Reference
       FSharpImplUtil.SetName(this, name);
 
     ITypeArgumentList ITypeArgumentOwner.TypeArgumentList => TypeArgumentList;
+    private string TypeArgumentsText => TypeArgumentList?.GetText() ?? "";
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/ReferenceExpr.cs:85]: Please construct the text properly, like it's done in QualifiedName; we don't want new lines, comments and other stuff in places we'd use it.

**Review by auduchinok — CHANGES_REQUESTED** [review]

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Highlightings/FSharpErrorUtil.fs (hunk 1 above)
Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Highlightings/FSharpErrorUtil.fs]: @auduchinok, added a bit of recursion

File: ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/ReferenceExpr.cs (hunk 2 above)
Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Psi/src/Impl/Tree/ReferenceExpr.cs]: We agreed to show the "Simplify lambda" tooltip for the case of "complex" qualified names

**Review by auduchinok — APPROVED** [review]

**Pull request merged**